### DISCUSSION TITLE
autokey and .ahk file

### DISCUSSION THREAD

**Discussion by ZeRenCZ** [autokey/autokey#614] [Q&A]
hi, I have combat mode for Guild Wars 2 and has .ahk file that is on windows opened by AutoHotkey.exe, but on linux it doesnt work (it should create icon on panel)

so could .ahk file be opened by Autokey same way?

**Comment by josephj11** [comment]
We see a number of people coming from Windows/AHK who could use a solution to this issue.

AHK is a feature rich application. It would be a major piece of work to implement (and maintain!) all of that for AutoKey.

Perhaps, a better approach would be to develop an external program that reads an AHK configuration file and converts some of the phrases and scripts to ones that work in AutoKey. If well designed, it could be an extensible framework so that more features and options could be added gradually. It could also do part of the work, building phrases and scripts that require more work, but are better than starting from scratch.

The major point of this is that it doesn't need to interact with AutoKey itself or modify our code base at all. This has several advantages.
* We have very limited developer resources and they wouldn't need to be involved.
* Since it would be external to AutoKey, it could be written using any language and tools its developers prefer.
* it doesn't have to be pretty or efficient because it would be a one-time conversion process, not something that runs all the time. 

Long ago, I wrote a proof of concept (very incomplete) bash script to convert AHK phrases to AutoKey phrases. It is available upon request, but it would have to be rewritten to make it really useful. For one thing, it can't handle multi-line phrases at all.

**Reply by ineuw** [reply]
Anything Autohotkey does, so can Python. If Python does, then so is Autokey, and this is the rule to apply.  I have a large collection of AHK keyboard macros, and read up on Python, (trial and error) and wrote corresponding Autokey macros for all of them. I also use Wine, though, I haven't tried to use Autokey to execute a Wine managed Windows software.

**Reply by josephj11** [reply]
@ineuw I can't recall specific issues ATM, but there have been reports of problems when using AutoKey with wine applications.

**Reply by tigerhawkvok** [reply]
4 years+ of thread necro here, but if you still have that script I'd love it. It's honestly the most painful part of using Linux right now, and even pretty boring. Mostly just [maps Emacs keyboard shortcuts](https://gist.githubusercontent.com/tigerhawkvok/5be684c160ff0cc710f5e117ac9e1877/raw/ba8de2d62a4c18fac9c3b9fb0514adc9883fd398/AutoHotKey.ahk) where I can't natively enable them.

I can always try feeding it through an LLM I suppose, but my gut is generally to go with something deterministic.

**Comment by josephj11** [comment]
Welcome to the AutoKey community, @tigerhawkvok! 

IDK AHK, but the script doesn't look too complicated on the surface.
Recoding one that long would be a labor of love, but it's probably doable. A lot of it looks like repeated templates. I would get a few bits of it to work at a time rather than trying to do the whole thing in one go.

The problem with LLMs is that there's very little AutoKey code out there to train on and so the attempts I've seen in the past were very poor. But if you can turn your code into a carefully written spec, it might work.

If you do get LLMs to work, please let us know how!

Otherwise,  we'd be happy to work with you to code it yourself - but on [Gitter](https://gitter.im/autokey/autokey) or on our [support group](https://groups.google.com/forum/#!forum/autokey-users), so it doesn't clutter up GitHub. Gitter is probably better because it supports markdown of code blocks to maintain indentation...
